kong/graphql_rate_limiting_cost_decoration_service.go

```diff
@@ -0,0 +1,152 @@
+package kong
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+type AbstractGraphqlRateLimitingCostDecorationService interface {
+	Create(ctx context.Context, costDeco *GraphqlRateLimitingCostDecoration) (*GraphqlRateLimitingCostDecoration, error)
+	Get(ctx context.Context, ID *string) (*GraphqlRateLimitingCostDecoration, error)
+	Update(ctx context.Context, costDeco *GraphqlRateLimitingCostDecoration) (*GraphqlRateLimitingCostDecoration, error)
+	Delete(ctx context.Context, ID *string) error
+	List(ctx context.Context, opt *ListOpt) ([]*GraphqlRateLimitingCostDecoration, *ListOpt, error)
+	ListAll(ctx context.Context) ([]*GraphqlRateLimitingCostDecoration, error)
+}
+
+type GraphqlRateLimitingCostDecorationService service
+
+func (s *GraphqlRateLimitingCostDecorationService) Create(
```

Comment:
How about adding go docs for those exported types and methods?

Reply:
Done!